solr/core/src/java/org/apache/solr/store/blob/metadata/CorePushPull.java

```diff
@@ -419,45 +433,51 @@ protected String pushFileToBlobStore(CoreStorageClient blob, Directory dir, Stri
+    private void logBlobAction(String action, long expectedFilesAffected, long actualFilesAffected, long bytesTransferred, boolean isSuccessful,
+        long requestQueuedTimeMs, int attempt, long startTimeMs) throws Exception {
       long now = System.nanoTime();
       long runTime = now - startTimeMs;
       long startLatency = now - requestQueuedTimeMs;
 
       String message = String.format(Locale.ROOT,
             "PushPullData=[%s] action=%s storageProvider=%s bucketRegion=%s bucketName=%s "
-              + "runTime=%s startLatency=%s bytesTransferred=%s attempt=%s filesAffected=%s localGeneration=%s blobGeneration=%s ",
+              + "runTime=%s startLatency=%s bytesTransferred=%s attempt=%s expectedFilesAffected=%s actualFilesAffected=%s isSuccessful=%s "
+              + "localGeneration=%s blobGeneration=%s ",
           pushPullData.toString(), action, coreStorageClient.getStorageProvider().name(), coreStorageClient.getBucketRegion(),
-          coreStorageClient.getBucketName(), runTime, startLatency, bytesTransferred, attempt, filesAffected,
-          solrServerMetadata.getGeneration(), blobMetadata.getGeneration());
+          coreStorageClient.getBucketName(), runTime, startLatency, bytesTransferred, attempt, expectedFilesAffected, actualFilesAffected,
+          isSuccessful, solrServerMetadata.getGeneration(), blobMetadata.getGeneration());
       log.info(message);
     }
 
     /**
      * Downloads files from the Blob store 
      * @param destDir (temporary) directory into which files should be downloaded.
      * @param filesToDownload blob files to be downloaded
+     * @return number of files downloaded and number of bytes transferred successfully
      */
     @VisibleForTesting
-    protected void downloadFilesFromBlob(Directory destDir, Collection<? extends BlobFile> filesToDownload) throws Exception {
+    protected Pair<Long, Long> downloadFilesFromBlob(Directory destDir, Collection<? extends BlobFile> filesToDownload) throws Exception {
       // Synchronously download all Blob blobs (remember we're running on an async thread, so no need to be async twice unless
       // we eventually want to parallelize downloads of multiple blobs, but for the PoC we don't :)
+      long filesDownloaded = 0;
+      long bytesTransferred = 0;
       for (BlobFile bf: filesToDownload) {
         log.info("About to create " + bf.getSolrFileName() + " for core " + pushPullData.getCoreName() +
             " from index on blob " + pushPullData.getSharedStoreName());
-        IndexOutput io = destDir.createOutput(bf.getSolrFileName(), DirectoryFactory.IOCONTEXT_NO_CACHE);
 
-        try (OutputStream outStream = new IndexOutputStream(io);
-          InputStream bis = coreStorageClient.pullStream(bf.getBlobName())) {
-          IOUtils.copy(bis, outStream);
-        }
+          try (IndexOutput io = destDir.createOutput(bf.getSolrFileName(), DirectoryFactory.IOCONTEXT_NO_CACHE);
```

Comment:
Moved IndexOutput initialization into the try with resources since it can throw an exception it also implements Closable

Reply:
Looked at the code and `OutputStream outStream = new IndexOutputStream(io)` was already in the try-with block and will end up closing the IndexOutput you moved. But the change has no affect if close if called twice so no change needed here, just wanted to note that.